filesource/discovery.go

```diff
@@ -3,261 +3,194 @@ package filesource
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"io/ioutil"
+	"math/rand"
 	"os"
+	"os/exec"
 	"regexp"
+	"time"
 
-	"github.com/estuary/connectors/schema_inference"
 	"github.com/estuary/flow/go/parser"
 	"github.com/estuary/flow/go/protocols/airbyte"
-	log "github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
 )
 
-// How many documents should we read from an individual file during discovery.
-const DISCOVER_DOC_LIMIT = 40000
-
-// Baseline document schema for resource streams we discover if we fail
-// during the schema-inference step.
-var DISCOVER_FALLBACK_SCHEMA = schema_inference.Schema(`{
-		"type": "object",
-		"properties": {
-			"_meta": {
-				"type": "object",
-				"properties": {
-					"file": { "type": "string" },
-					"offset": {
-						"type": "integer",
-						"minimum": 0
-					}
-				},
-				"required": ["file", "offset"]
-			}
-		},
-		"required": ["_meta"]
-	}`)
+// The maximum amount of time we'll read documents for.
```

Comment:
If I'm understanding the code properly it looks like we could actually take up to `2*discoverBudget` for discovery (once for the listing to time out, then another time for the reading). Is that right?.

Perhaps this should be mentioned in the comment, or else have separate `discoverListingBudget` and `discoverReadBudget` constants?